python/createCollectionDownload.py

```diff
@@ -1,76 +1,209 @@
-    # ONCE YOU HAVE A URL FROM THE CHAINLISTDOWNLOADS ENDPOINT, THEN YOU'RE GOOD TO GO AND CAN DO WHATEVER YOU NEED WITH THAT FILE
-
-    fileGenerated = False
-    createdCollectionFileURL = False
+    print(f"Checking for status of downloaded file...")
     
-    while(fileGenerated == False):
-        print("Checking for status of generated file " + str(collection_download_id) + "...")
-        response = requests.get(url='https://location.chainxy.com/api/Downloads/{}'.format(collection_download_id), headers=headers)
-        r_body = json.loads(response.text)['Record']
+    while(file_generated == False):
+        
+        r = request_api(url=f'https://location.chainxy.com/api/Downloads/{collection_download_id}', cxy_api_key=cxy_api_key)
+        r_body = json.loads(r.text)['Record']
 
         if r_body['Status'] == 0:
-            print("Download " + str(collection_id) + " is still generating")
+            print(f"Download {str(collection_download_id)} is still generating...")
             time.sleep(5)
 
         elif r_body['Status'] == 2:
-            print("File generation failed. Speak to ChainXY for assistance")
-            fileGenerated = True
+            print("File generation failed. Please reach out to ChainXY support for assistance.")
+            file_generated = True
 
         elif r_body['Status'] == 1:
             print("File generation completed!")
-            fileGenerated = True
-            createdCollectionFileURL = r_body['Link']
+            file_generated = True
+            collection_download_link = r_body['Link']
+    
+    return collection_download_link
+
+def download_collection(cxy_api_key:str, collection_id:int, collection_type:str, cache_time:int = 24):
+    """
+    Downloads a collection based on the provided collection ID and collection type. An optional input cache_time determines if a new download should be made.
+    Params:
+        cxy_api_key:str - ChainXY API Key
+        collection_id:str - ID of the collection for which a new download will be initiated.
+        collectiontype:str - 'chain' or 'center' collection
+        cache_time:int - max. time in hours since the latest download before a new one is requested
+    Returns:
+        collection_download_link:str - link to S3 URL of the downloaded collection
+    """
+
+    validate_inputs(cxy_api_key, collection_type, cache_time)    
+    print(f"""Preparing collection download with inputs:
+        {collection_id=}
+        {collection_type=}
+        {cache_time=}
+        """)
+
+    ## GET THE APPROPRIATE API REQUEST URLS FOR COLLECTION TYPE
+    records_url = check_url = new_download_url = ''
+    if collection_type == 'chain':
+        records_url = f'https://location.chainxy.com/api/Downloads?fields=Id,ChainList,Label,User,ReportType,CreateDate,DataDate,Status,Format,Count,Link,DisplayValue,&Query=%7B%22ChainListId%22:{collection_id}%7D&OrderBy=-CreateDate'
```

Comment:
we need to make sure that we're filtering out other report types when we search for downloads here, when I ran this it picked up a ChangesOverTime report, even though it's only meant to pick up ChainLocationReports. See ReportType enum in the backend.